java/datastructures/lib/src/test/java/datastructures/linkedlist/LinkedListTest.java

```diff
@@ -118,4 +126,93 @@ void insertAfterNodeTEST() {
     assertTrue(sut.includes(15));
     System.out.println("Was 15 present in the list: " + sut.includes(15));
   }
+
+  //////////// CODE CHALLENGE 07 ////////////////////////
+
+  @Test
+  @DisplayName("K Same Length TEST")
+  void kSameLengthTEST() {
+    LinkedList sut = new LinkedList();
+    //Act
+    sut.insert(2);
+    sut.insert(8);
+    sut.insert(3);
+    sut.insert(1);
+
+    int k = 3;
+    int expectedValue = 1;
+    //Assert
+    assertEquals(sut.kthFromEnd(k), expectedValue);
+    System.out.println("We were looking for " + sut.kthFromEnd(k) + " and got back " + expectedValue + ".");
+  }
+
+  @Test
+  @DisplayName("K longer than length")
```

Comment:
If you want to test whether your method throws exceptions when expected, you can call your method inside JUnit's `assertThrows` assertion:

`Exception exception = assertThrows(IllegalArgumentException.class, () -> {
      sut.kthFromEnd(k);
    });`

This will create an Exception object called `exception`. Use `exception.getMessage()` to get the error message (in this case, "k exceeds length"), then compare that string to the expected error message using `assertEquals()`.